db/blip_sync_context.go

```diff
@@ -260,6 +266,11 @@ func (bsc *BlipSyncContext) handleChangesResponse(sender *blip.Sender, response
 
 			revSendTimeLatency += time.Since(changesResponseReceived).Nanoseconds()
 			revSendCount++
+
+			// TODO: Await a successful response before calling postSendRevisionCallback
```

Comment:
It looks like there are a number of returns between here and preSendRevisionCallback.  Will those returns block checkpointing (i.e. do we need handling to remove a sequence from checkpointer.expectedSeqs if we hit error conditions?  Actually, should we be just be waiting to increment expectedSeqs until line 268, when we've confirmed a successful send?

Reply:
Good catch. Yep that would certainly cause problems. I agree with the suggestion of adding to expectedSeqs once we've sent the request.